### ISSUE TITLE
warnings cannot be accessed in notebooks

### ISSUE THREAD

**Issue by yueyinqiu** [dotnet/TorchSharp#1307]
For example, the following codes can produce a warning in the console

```csharp
using TorchSharp;

var loss = torch.nn.MSELoss();
var x = torch.zeros([]);
var y = torch.zeros([1]);
loss.call(x, y);
```

![image](https://github.com/dotnet/TorchSharp/assets/18749772/2411f092-81bf-4ed4-9b17-2b479ffea075)

However this seems to be unable to access on other platforms, like polyglot notebooks.

**Comment by NiklasGustafsson** [comment]
Right. The warning is coming from libtorch, and going to stderr. I don't know if there's a way to redirect errors to the Notebook console, but that would be the general solution.

**Comment by yueyinqiu** [comment]
To be honest, I’m not entirely sure if this is an issue that should be resolved within TorchSharp itself.

If we consider outputting warnings to stderr as the expected behavior, then this issue is actually caused by Polyglot Notebooks not displaying stderr correctly.

On the other hand, we could also consider whether TorchSharp should implement a dedicated warning mechanism (e.g., a custom logger or diagnostic API) instead of writing directly to stderr.
